bench/src/main/scala/parsley/bench/JsonBench.scala

```diff
@@ -0,0 +1,51 @@
+package parsley.bench
+
+import java.util.concurrent.TimeUnit
+import org.openjdk.jmh.annotations._
+import org.typelevel.jawn.ast._
+import parsley._, Combinator._, Parsley._
+import scala.io.Source
+
+/* Based on https://github.com/typelevel/jawn/blob/v1.0.0/benchmark/src/main/scala/jawn/JmhBenchmarks.scala */
+@State(Scope.Benchmark)
+@BenchmarkMode(Array(Mode.AverageTime))
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+abstract class JmhBenchmarks(name: String) {
+  val text: String =
+    Source.fromResource(name, getClass.getClassLoader).getLines().mkString("\n")
+
+  def json: Parsley[JValue] = {
+    val jsontoks = LanguageDef("", "", "", false, NotRequired, NotRequired, NotRequired, NotRequired, Set.empty, Set.empty, true, Predicate(Char.isWhitespace))
+    val tok = new TokenParser(jsontoks)
+    lazy val obj: Parsley[JValue] = tok.braces(tok.commaSep(+(tok.stringLiteral <~> tok.colon *> value)).map(pairs => JObject.fromSeq(pairs)))
```

Comment:
By the way, since this parser was originally written, `+` is pointless. I should have deprecated it really (or removed it outright). The optimisation that it is supposed to perform is now down automatically by the Parsley compiler.